airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -933,16 +934,17 @@ def list_by_timespan(
                     delimiter=delimiter,
                     versions=versions,
                 )
-                list(blobs)
+
+            blob_names = [
+                blob.name
+                for blob in blobs
+                if timespan_start <= blob.updated.replace(tzinfo=timezone.utc) < timespan_end
+            ]
```

Comment:
This PR looks great and returns the provider to it's original behaviour, it's working fine for me when testing locally.

It's worth noting though that when a delimiter is passed in, the filter here has no effect. It will simply return all files. The filtering only works when `match_glob` is used. 

This is how the provider has been working up until this point however so this can be addressed later, if ever as delimiter is deprecated.